framework/src/headers/ip/v6/mod.rs

```diff
@@ -0,0 +1,312 @@
+   +                                                               +
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                                                               |
+   +                                                               +
+   |                                                               |
+   +                      Destination Address                      +
+   |                                                               |
+   +                                                               +
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+   Version              4-bit Internet Protocol version number = 6.
+
+   Traffic Class        8-bit traffic class field.
+
+   Flow Label           20-bit flow label.
+
+   Payload Length       16-bit unsigned integer.  Length of the IPv6
+                        payload, i.e., the rest of the packet following
+                        this IPv6 header, in octets.  (Note that any
+                        extension headers present are considered part of
+                        the payload, i.e., included in the length count.)
+
+   Next Header          8-bit selector.  Identifies the type of header
+                        immediately following the IPv6 header.  Uses the
+                        same values as the IPv4 Protocol field [RFC-1700
+                        et seq.].
+
+   Hop Limit            8-bit unsigned integer.  Decremented by 1 by
+                        each node that forwards the packet. The packet
+                        is discarded if Hop Limit is decremented to
+                        zero.
+
+   Source Address       128-bit address of the originator of the packet.
+
+   Destination Address  128-bit address of the intended recipient of the
+                        packet (possibly not the ultimate recipient, if
+                        a Routing header is present).
+*/
+
+// V6 addresses are 128 bits wide.
+pub type Rawv6Address = u128;
+
+#[derive(Default)]
+#[repr(C, packed)]
+pub struct Ipv6Header {
+    version_to_flow_label: u32,
+    payload_len: u16,
+    next_header: u8,
+    hop_limit: u8,
+    src_ip: Rawv6Address,
+    dst_ip: Rawv6Address,
+}
+
+// IPv6 can encapsulate any L4 IP protocol.
+impl IpHeader for Ipv6Header {}
+
+impl EndOffset for Ipv6Header {
+    // Note this does not allow IPv6-in-IPv6 or any other tunneling variants.
+    type PreviousHeader = MacHeader;
+
+    #[inline]
+    fn offset(&self) -> usize {
+        // IPv6 Header is always 40 bytes: (4 + 8 + 20 + 16 + 8 + 8 + 128 + 128) / 8 = 40
+        40
+    }
+
+    #[inline]
+    fn size() -> usize {
+        // Struct is always 40 bytes as well
+        40
+    }
+
+    #[inline]
+    fn payload_size(&self, _: usize) -> usize {
```

Comment:
Style wise would it be clearer to use `_hint` rather than just `_`.

Reply:
Addressed in 70224de